llvm/include/llvm/Support/TimeProfiler.h

```diff
@@ -181,9 +187,18 @@ class TimeTraceScope {
       Entry = timeTraceProfilerBegin(Name, Detail);
   }
   TimeTraceScope(StringRef Name,
```

Comment:
We should not use the `TimeTraceScope` for instant events at all.
It does not make much sense to deal with the complexities of object lifetimes, the instant events can be modelled with a single function call.

This should also allow to treat them separately in implementation and simplify the code (e.g. there's no need for pairing the `begin`/`end` calls and we can immediately attribute the instant event to the durable event on the top of the stack, if we want time granularity to have some effect on instant events too).

Reply:
Done.